src/nodes/primitive/range.ts

```diff
@@ -104,39 +139,64 @@ export const rangeNode = defineNodeKind<RangeNode>(
                         bound,
                         `${size} ${
                             bound.comparator === "==" ? "===" : bound.comparator
-                        } ${bound.limit}`,
+                        } ${
+                            isDate(bound.limit)
+                                ? bound.limit.valueOf()
+                                : bound.limit
+                        }`,
                         ctx
                     )
                 )
                 .join("\n")
         },
         intersect: (l, r): RangeNode | Disjoint => {
+            if (typeof l.value !== typeof r.value) {
+                return Disjoint.from("range", l, r)
+            }
+            const leftValue = isDate(l.value) ? l.value.valueOf() : l.value
```

Comment:
I thought the point of `numericMin` and `numericMax` was to avoid having to create separate variables like this? We'd just use those for everything.

I also think the usage of value at the top level here is quite confusing- it's as if it's something fundamental to the range but it's actually just the first limit? Maybe was being used as heuristic for the type of both, but again I'd remove it, try to use numericMin/Max, and add a `rangeKind` if needed